src/v/archival/ntp_archiver_service.cc

```diff
@@ -2739,6 +2701,68 @@ ntp_archiver::delete_segment(const remote_segment_path& path) {
     co_return res;
 }
 
+ss::future<size_t>
+ntp_archiver::delete_segments(const std::vector<lw_segment_meta>& lw_metas) {
```

Comment:
Have you considered moving the batching logic into `cloud_storage::remote::delete_objects`? Currently, other places duplicate the batching logic (`remote_partition::erase` and `clear_recovery_results`).

Reply:
I have considered moving matching into delete_objects, yes. But doing that requires answering some additional non-trivial questions.

First, the level of abstraction for `remote` if a bit vague, some methods operate on segments, others on objects. `delete_objects` operates on object level and knows only about keys. So, it can return information about key level operations (e.g. 10 successful deleted keys). With that we lose the ability to update metrics like "successfully deleted segments". We also lose the ability to log in time the segment names we are deleting (we have to either log everything before calling delete_objects, or log in delete objects the physical key names we are deleting.

I see two paths for this first problem: 1. give up on logging and exact metrics, 2. introduce a way to correlate per-key responses to source segments. A comment here might be that right now neither logging, nor metrics are very precise anyway so first approach wouldn't be really a problem.

Until I have a better context for making decisions I decided to provide a ready solution for the immediate problem, and separately, work on a more fundamental solution (including batching for ABS/GCS).